### DISCUSSION TITLE
cgroup support

### DISCUSSION THREAD

**Discussion by tubby1981** [FoobarOy/foomuuri#41] [General]
Is cgroup supported?

I now use it to route outgoing traffic with IP Tables and route tables
`iptables -t mangle -A OUTPUT -p tcp --dport 80 -m cgroup --cgroup "${classid}" -j MARK --set-mark "${classid}"
iptables -t mangle -A OUTPUT -p tcp --dport 443 -m cgroup --cgroup "${classid}" -j MARK --set-mark "${classid}"`

**Comment by kimheino** [comment]
I just added it:

https://github.com/FoobarOy/foomuuri/commit/13939c653f91f392b02a29c139fda9c39ddf5a52

**Comment by tubby1981** [comment]
Thanks, i will test it soon :+1:

**Reply by tubby1981** [reply]
Was trying to build a deb package on Debian but got the following error.

```../src/foomuuri --set=etc_dir=60-conntrack --set=share_dir=../etc --set=state_dir=60-conntrack --set=run_dir=60-conntrack check
60-cgroup/next.fw:179:27-53: Error: cgroupv2 path fails: No such file or directory
                socket cgroupv2 level 2 "system.slice/sshd.service" accept
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
60-cgroup/next.fw:194:27-53: Error: cgroupv2 path fails: No such file or directory
                socket cgroupv2 level 2 "system.slice/sshd.service" accept
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Nftables failed to check ruleset, error code 1
```

On Debian we need. Can that be changed/fixed for example in the Makefile?
```
systemctl show -p ControlGroup sshd
ControlGroup=/system.slice/ssh.service
```

**Comment by kimheino** [comment]
Commit afecf41 fixes this. On Fedora it's `sshd.service` and on Debian it's `ssh.service`. This one test fails when running as root and sshd is not running, or doesn't exists. It's success if run as normal user.

It's better to not require any specific service running on tests., even if it makes tests weaker.

**Comment by kimheino** [comment]
I reverted above commit and made a better one: b5cb326

Debian's reprotest had one more failure on this test. Not even the system.slice exist.
